### PR TITLE
Make automountserviceaccounttoken configurable

### PR DESCRIPTION
**Pull request by LinAnt**
The service account token is needed if you fetch secrets from external sources (i.e. vault) and thus it needs to be configurable.

### CODE REVIEW

**Comment by LinAnt** [comment]
Is there anything blocking this PR or something that needs to be improved?

**Comment by ywk253100** [comment]
@LinAnt How about exposing these configuration items in `values.yaml` and updating the`READEME.md` so that users can find them easily?

**Comment by LinAnt** [comment]
I can do that :+1:

**Review by ywk253100 — APPROVED** [review]

**Comment by ywk253100** [comment]
@LinAnt Thanks a lot. Could you squash the commits into one?

**Review by reasonerjt — APPROVED** [review]

**Comment by reasonerjt** [comment]
@LinAnt Could you please rebase the branch so it can be easily merged?

Thanks in advance!

**Comment by LinAnt** [comment]
My editor decided to format the README.md table while rebasing.

I fixed the typos i.e. "natary" -> "notary" and then trivy.priorityClassName was missplaced so it won't render on master atm. 

Both things are fixed now :+1:

**Comment by LinAnt** [comment]
exporter.replicas was also malformed, so took care of that as well :+1:

**Comment by ywk253100** [comment]
@LinAnt Thanks a lot!

File: README.md [hunk 1]
```diff
@@ -87,286 +93,299 @@ helm uninstall my-release
...
+| `registry.registry.resources`                                               | The [resources] to allocate for container                                                                                                                                                                                                                                                                                                                                                                                                                                                      | undefined                                                                           |     | `dev` |
```
Comment (ywk253100) [inline README.md:257]: There is redundant `dev` in the last column

File: README.md [hunk 2]
```diff
@@ -87,286 +93,299 @@ helm uninstall my-release
...
+| --------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- | --- | ----- |
```
Comment (ywk253100) [inline README.md:97]: Seems there are two redundant columns?

Comment (LinAnt) [reply README.md:97]: Yep, not sure what the `dev` and `1` are there for, seems to be some weird formatting from the past?

image

columns do not match on master either, should I just remove the extra `1` and `dev`

Comment (LinAnt) [reply README.md:97]: Seems the `dev` blocks are from imagetag lines. fixing that :+1:

Comment (LinAnt) [reply README.md]: the rouge `1`s were amount of replicas, fix pushed

**Pull request merged**